adoc-mode.el

```diff
@@ -3416,16 +3487,32 @@ LOCAL-ATTRIBUTE-FACE-ALIST before it is looked up in
   "Keymap used in adoc mode.")
 
 
+(defcustom adoc-mode-font-lock-keywords
```

Comment:
Why does this have the same name as the variable that you set above? Seems incorrect to me.

Reply:
Variable `adoc-mode-font-lock-keywords` is used with `(setq adoc-mode-font-lock-keywords...)` but not defined. I don't see code like this:

```emacs-lisp
(defvar adoc-mode-font-lock-keywords ...)
```

I used example from `dns-mode.el` for writing this code. Also I saw some TODO's like this:

> 1539 ;; TODO: use same regexps as for font lock

I think, it is a good idea replace functions for regexp to constants with more human readable (and maintain!) code, based on `rx`.

Reply:
I think you didn't get my point - this defcustom has some value that simple gets overwritten here:

```
(setq adoc-mode-font-lock-keywords (adoc-get-font-lock-keywords))
```

I'm fine with simplifying code by using `rx`, but I just don't understand what's the point of this `defcustom` right now.

Reply:
Maybe, I add before `adoc-calc` function declaration like this?

```emacs-lisp
(defvar adoc-font-lock-keywords nil "Keywords for font-lock in `adoc-mode'.")
```

Reply:
I'd probably remove this `adoc-calc` completely, as few modes do something like this. If you want to factor in some font-lock customizations you typically just disable/enable the mode and its init code will take care of such updates. In general it's very bad style to have a top-level function call to initialize anything (as `adoc-mode` currently does).